### PR TITLE
[BACKLOG-35252] Initial implementation of lineage for LDOS/LDC based on

### PR DESCRIPTION
**Pull request by peterrinehart**
prototype from Angel Ramos Cardona.

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 1m
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 47s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 1m 12s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/metaverse/graph/GraphCatalogWriter.java [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+            dataResource.setDbName( dbNode.getProperty( DictionaryConst.PROPERTY_DATABASE_NAME ) );
+          }
+        }
+      }
+    }
+  }
+
+  private String getSourceName( String fullName ) {
+    String sourceName = null;
+    if ( fullName.contains( "/" ) ) {
```
Comment (ricksjames) [inline core/src/main/java/org/pentaho/metaverse/graph/GraphCatalogWriter.java:194]: Does the forward slash need to platform dependent?

File: core/src/main/java/org/pentaho/metaverse/graph/GraphCatalogWriter.java [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+                      .out( DictionaryConst.LINK_CONTAINS )
+                      //.has( DictionaryConst.PROPERTY_NAME, Tokens.T.eq, "SUR" )
+                      .cast( Vertex.class );
+      // returns all fields from all previously identified output tables
+      GremlinPipeline<Graph, Vertex> tableFieldsPipe =
+        new GremlinPipeline<Graph, Vertex>( graph )
+          .V()
+          .has( DictionaryConst.PROPERTY_TABLE, Tokens.T.eq, outputTarget.getName() )
+          .out( DictionaryConst.LINK_CONTAINS )
+          //.has( DictionaryConst.PROPERTY_NAME, Tokens.T.eq, "SUR" )
```
Comment (ricksjames) [inline core/src/main/java/org/pentaho/metaverse/graph/GraphCatalogWriter.java:264]: remove unused code

File: core/src/main/java/org/pentaho/metaverse/graph/GraphCatalogWriter.java [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+   */
+  private void linkTargetFieldsToSources( List<LineageDataResource> outputTargets, List<LineageDataResource> inputSources, Graph graph ) {
+    for ( LineageDataResource outputTarget : outputTargets ) {
+      // returns all fields from previously identified output files
+      GremlinPipeline<Graph, Vertex> fileFieldsPipe =
+              new GremlinPipeline<Graph, Vertex>( graph )
+                      .V()
+                      .has( DictionaryConst.PROPERTY_PATH, Tokens.T.eq, outputTarget.getPath() )
+                      .out( DictionaryConst.LINK_CONTAINS )
+                      //.has( DictionaryConst.PROPERTY_NAME, Tokens.T.eq, "SUR" )
```
Comment (ricksjames) [inline core/src/main/java/org/pentaho/metaverse/graph/GraphCatalogWriter.java:256]: remove unused code

File: core/src/main/java/org/pentaho/metaverse/graph/catalog/LineageDataResource.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+      .concat( " path: " ).concat( path )
+      .concat( " catalogResourceId: " ).concat( catalogResourceID )
+      .concat( " dbHost: " ).concat( dbHost )
+      .concat( " dbPort: " ).concat( dbPort )
+      .concat( " dbSchema: " ).concat( dbSchema );
+  }
+
+  @Override
+  public boolean equals( Object o ) {
+    if ( o instanceof LineageDataResource ) {
```
Comment (ricksjames) [inline core/src/main/java/org/pentaho/metaverse/graph/catalog/LineageDataResource.java:122]: Same here, should document in comment why some fields are skipped in the equals comparison. Especially given the hashCode includes them.

File: core/src/main/java/org/pentaho/metaverse/graph/GraphCatalogWriter.java (hunk 1 above)
Comment (peterrinehart) [reply core/src/main/java/org/pentaho/metaverse/graph/GraphCatalogWriter.java:194]: I don't know; I believe it depends on how Windows-stored files would be indexed in the catalog.

**Comment by buildguy** [comment]
### :warning: Build finished in 3m 20s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.metaverse.locator.DIRepositoryLocatorTest.testStopLocatorScan</b> (click to expand)</summary><p>

```
${result.errorDetails}
```

</p></details>

  
Tests run: **890**, Failures: **1**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/metaverse-plugin/job/PR-644/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by ricksjames — APPROVED** [review]

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 4m 14s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **891**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/metaverse-plugin/job/PR-644/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by peterrinehart** [comment]
new PR opened to address comments but keep edit history locally

**Pull request closed without merging**